src/rank_llm/rerank/rank_gpt.py

```diff
@@ -56,9 +59,9 @@ def __init__(
             keys = [keys]
         if not keys:
             raise ValueError("Please provide OpenAI Keys.")
-        if prompt_mode not in [PromptMode.RANK_GPT, PromptMode.LRL]:
+        if prompt_mode not in [PromptMode.RANK_GPT, PromptMode.RANK_GPT_APEER, PromptMode.LRL]:
             raise ValueError(
-                f"unsupported prompt mode for GPT models: {prompt_mode}, expected {PromptMode.RANK_GPT} or {PromptMode.LRL}."
+                f"unsupported prompt mode for GPT models: {prompt_mode}, expected {PromptMode.RANK_GPT}, {PromptMode.RANK_GPT_APEER} or {PromptMode.LRL}."
```

Comment:
why do we need this format? is it better than rank_gpt or different usage?

Reply:
Another variant from here - https://arxiv.org/abs/2406.14449

Not too much work to add support, more people will stick to our framework for running that additional model too.